### PR TITLE
mgr/dashboard: Re-enable OSD's table autoReload

### PR DESCRIPTION
**Pull request by tspmelo**
Fixes: https://tracker.ceph.com/issues/45303

Signed-off-by: Tiago Melo <tmelo@suse.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by tspmelo** [comment]
jenkins test dashboard

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/shared/device-list/device-list.component.ts [hunk 1]
```diff
@@ -41,7 +41,7 @@ export class DeviceListComponent implements OnInit {
     private osdService: OsdService
   ) {}
 
-  ngOnInit() {
+  ngOnChanges() {
     const updateDevicesFn = (devices: CdDevice[]) => (this.devices = devices);
```
Comment (bk201) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/shared/device-list/device-list.component.ts:45]: Would it be better to keep column init in `ngOnInit`?

**Comment by tspmelo** [comment]
jenkins test docs

**Comment by tspmelo** [comment]
jenkins test dashboard

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/shared/device-list/device-list.component.ts (hunk 1 above)
Comment (bk201) [reply src/pybind/mgr/dashboard/frontend/src/app/ceph/shared/device-list/device-list.component.ts]: Thanks for addressing the comment. Tested again and it works!

**Comment by tspmelo** [comment]
jenkins retest this please

**Comment by tspmelo** [comment]
jenkins retest this please

**Comment by tspmelo** [comment]
jenkins test make check

**Comment by tspmelo** [comment]
jenkins test make check

**Comment by tspmelo** [comment]
jenkins test make check

**Pull request merged**